### ISSUE TITLE
Microsoft.Azure.Functions.Worker.Extensions The target framework 'net8.0' must be 'net6.0'

### ISSUE THREAD

**Issue by MikeYeager** [Azure/azure-functions-dotnet-worker#2489]
### Description

I have a C# Azure Function project set to target net8.0. When I build, I get the error message: The target framework 'net8.0' must be 'net6.0'. When I click on the error message in VS, it opens WorkerExtension.csproj in a temp folder. In that file I find this:

```
    <Target Name="_VerifyTargetFramework" BeforeTargets="Build">
        <!-- It is possible to override our TFM via global properties. This can lead to successful builds, but runtime errors due to incompatible dependencies being brought in. -->
        <Error Condition="'$(TargetFramework)' != 'net6.0'" Text="The target framework '$(TargetFramework)' must be 'net6.0'. Verify if target framework has been overridden by a global property." />
    </Target>
```
These are the NuGet packages and versions:
![image](https://github.com/Azure/azure-functions-dotnet-worker/assets/7063813/5f556804-cb15-4a6f-9fbb-124bb240b126)

### Steps to reproduce

Create a function app with these NuGet packages, set it to target net8.0, build.

**Comment by satvu** [comment]
This seems similar to https://github.com/Azure/azure-functions-dotnet-worker/issues/2316 - does updating to the latest SDK preview package fix your issue?

**Comment by MikeYeager** [comment]
You are correct. I swear I searched for related issues before I posted, but did not come across this one for some reason. Thank you.

**Comment by OlufTE** [comment]
@satvu Do you know when the fix is expected to be released? I am in the middle of migrating a larger code base to the isolated worker model, and it is an issue that there is no released version of the SDK that actually works (earlier versions than 1.17.2 have other issues).

**Comment by mbrown-dlb** [comment]
Facing this same issue with a pipeline running on Azure DevOps. I hate to have to modify all of my pipelines to pull the SDK because the base image doesn't have the latest official versions on it that work.
